skglm/utils.py

```diff
@@ -254,3 +255,51 @@ def check_group_compatible(obj):
                 f"'{obj_name}' is not block-separable. "
                 f"Missing '{attr}' attribute."
             )
+
+
+spec_AndersonAcceleration = [
+    ('K', int32),
+    ('n_features', int32),
+    ('current_iter', int32),
+    ('arr_w', float64[:, ::1]),
+]
+
+
+@jitclass(spec_AndersonAcceleration)
+class AndersonAcceleration:
+    """Anderson Accelerator."""
```

Comment:
If confirmed, this shall replace the acceleration procedure in other solvers.

Reply:
I don't think AA should be exposed this way. Also, see how we handle the singular matrix case. No need to catch `Exception("Singular matrix")`, you already have a `LinAlgError` raised by ` numpy` when the matrix is singular.
I wouldn't put AA in a utils file. If we keep this design, I'd put it in a `common` submodule.

 I agree that we could encapsulate the AA logic in some generic function / class to avoid redundancy though. @mathurinm  WDYT?

Reply:
I am also not a huge fan of having an AA class, but no strong opinion.

What's the benefit in your opinion Badr ?

Reply:
AA accelerate algorithms without interfering with how they work, also it's the same for all of them.
Therefore, I think we better separate the logic behind it instead of re-inviting it for every use case.

I opted for a class instead of functions simply because AA holds an internal state (the past K+1 iterates). Also, I ensured that we benefit from jit compilation by implementing it as ``numba`` jit class.

With that being implemented, boosting an algorithm with AA with be as simple as 
1. initializing an instance of the class
2. calling its ``explotate`` method within the loop of the algorithm.